lib/array_equals.rb

```diff
@@ -1,5 +1,23 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    if array1 == nil && array2 == nil
+      return true
+    else
+      return false
+    end
+  elsif array1.length != array2.length
+    return false
+  else
+    length = array1.length
+    index = 0
+    length.times do |number|
```

Comment:
The `times` loop takes care of tracking the index for you. You can substitute `number` for `index` in your conditional on line 16 and remove the index variable.

Reply:
True, thank you! Do you think it would make sense to name that iteration variable index or i instead of number, then?

Reply:
Yes. I think either `i` or `index` would be better than `number`.